crates/ruff_python_parser/src/parser/expression.rs

```diff
@@ -1933,8 +1933,16 @@ impl<'src> Parser<'src> {
     fn parse_comprehension(&mut self) -> ast::Comprehension {
         let start = self.node_start();
 
-        let is_async = self.eat(TokenKind::Async);
-        self.bump(TokenKind::For);
+        let is_async = if self.eat(TokenKind::Async) {
+            // test_err comprehension_missing_for_after_async
+            // (async)
+            // (x async x in iter)
+            self.expect(TokenKind::For);
+            true
```

Comment:
`(async` still panics which I don't think is related.

Reply:
I assume you plan to tackle this as a separate PR

Reply:
Yes

Reply:
(btw, this was fixed indirectly)